storm/src/main/storm/mesos/util/MesosCommon.java

```diff
@@ -105,7 +105,7 @@ public static String taskId(String nodeid, int port) {
   }
 
   public static String supervisorId(String nodeid, String topologyId) {
-    return String.format("%s-%s", nodeid, topologyId);
+    return String.format("%s%s%s", nodeid, DEFAULT_MESOS_COMPONENT_NAME_DELIMITER, topologyId);
```

Comment:
Shouldn't this be a call to `MesosCommon.getMesosComponentNameDelimiter()` instead of using `DEFAULT_MESOS_COMPONENT_NAME_DELIMITER` directly?  I realize it might be annoying to propagate the `Map conf` and `TopologyDetails info` through this call, so it might not be worth it.